user/user.go

```diff
@@ -26,22 +28,51 @@ type UserDetail struct {
 	Id            string   //no tag as we aren't getting it from json
 	Name          string   `json:"username"`
 	Emails        []string `json:"emails"`
+	Roles         []string `json:"roles"`
 	TermsAccepted string   `json:"termsAccepted"`
 	Pw            string   `json:"password"`
 	Verified      bool     `json:"authenticated"` //tag is name `authenticated` for historical reasons
 }
 
+const (
+	//predefined roles that tidepool will understand
+	CLINIC_ROLE = "clinic"
+)
+
 var (
 	User_error_name_pw_required = errors.New("User: both the name and pw are required")
 	User_error_no_details_given = errors.New("User: no user details were sent")
+	User_error_invalid_role     = errors.New("User: trying to set the role as")
+	AllowedUserRoles            = map[string]string{
+		CLINIC_ROLE: CLINIC_ROLE,
+	}
 )
 
+func validateRoles(roles []string) error {
+	for i := range roles {
+		if _, ok := AllowedUserRoles[roles[i]]; !ok {
+			configuredRoles := []string{}
+			for k := range AllowedUserRoles {
+				configuredRoles = append(configuredRoles, k)
+			}
+			return errors.New(fmt.Sprintf("User: trying to set role as `%s` which is not one of the valid roles `%s`", roles[i], configuredRoles))
+		}
+	}
+	return nil
+}
+
 func NewUser(details *UserDetail, salt string) (user *User, err error) {
 
 	if details.Name == "" || details.Pw == "" {
 		return nil, User_error_name_pw_required
 	}
 
+	//if yours are set ensure it is an allowed type
+	err = validateRoles(details.Roles)
```

Comment:
We cannot have roles set on a new user API call.  This is a security hole that would allow any user to give themselves any role.